react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+class AppDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isMenuClicked: false };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    event.preventDefault();
+    const targetClassList = event.target.classList;
+    for (let i = 0; i < targetClassList.length; i++) {
+      if (targetClassList[i] === 'menu-icon') {
+        this.setState({ isMenuClicked: true });
+      } else if (targetClassList[i] === 'container') {
+        this.setState({ isMenuClicked: false });
+      }
+    }
+    if (event.target.tagName === 'A') {
+      this.setState({ isMenuClicked: false });
+    }
```

Comment:
A simpler approach to this would be to just toggle the `isMenuClicked` property in state. If the menu is open close it, if it is closed open it.

```JSX
this.setState({
  isMenuClicked: !this.state.isMenuClicked
});
```